gymnasium/envs/toy_text/taxi.py

```diff
@@ -24,89 +24,113 @@
 
+    Destination on the map are represented with the first letter of the color.
+
+    Passenger locations:
+    - 0: Red
+    - 1: Green
+    - 2: Yellow
+    - 3: Blue
+    - 4: In taxi
+
+    Destinations:
+    - 0: Red
+    - 1: Green
+    - 2: Yellow
+    - 3: Blue
+
+    An observation is returned as an `int()` that encodes the corresponding state, calculated by
+    `((taxi_row * 5 + taxi_col) * 5 + passenger_location) * 4 + destination`
+
     Note that there are 400 states that can actually be reached during an
     episode. The missing states correspond to situations in which the passenger
     is at the same location as their destination, as this typically signals the
     end of an episode. Four additional states can be observed right after a
     successful episodes, when both the passenger and the taxi are at the destination.
     This gives a total of 404 reachable discrete states.
 
-    Each state space is represented by the tuple:
-    (taxi_row, taxi_col, passenger_location, destination)
+    ## Starting State
+    The episode starts with the player in a random state.
+
+    ## Rewards
+    - -1 per step unless other reward is triggered.
+    - +20 delivering passenger.
+    - -10  executing "pickup" and "drop-off" actions illegally.
+
+    An action that results a noop, like moving into a wall, will incur the time step
+    penalty. Noops can be avoided by sampling the `action_mask` returned in `info`.
 
-    An observation is an integer that encodes the corresponding state.
-    The state tuple can then be decoded with the "decode" method.
+    ## Episode End
+    The episode ends if the following happens:
 
-    Passenger locations:
-    - 0: R(ed)
-    - 1: G(reen)
-    - 2: Y(ellow)
-    - 3: B(lue)
-    - 4: in taxi
+    - Termination:
+            1. The taxi drops off the passenger.
 
-    Destinations:
-    - 0: R(ed)
-    - 1: G(reen)
-    - 2: Y(ellow)
-    - 3: B(lue)
+    - Truncation (when using the time_limit wrapper):
+            1. The length of the episode is 200.
 
-    ## Info
+    ## Information
 
-    ``step`` and ``reset()`` will return an info dictionary that contains "p" and "action_mask" containing
-        the probability that the state is taken and a mask of what actions will result in a change of state to speed up training.
+    `step()` and `reset()` return a dict with the following keys:
+    - p - transition proability for the state.
+    - action_mask - if actions will cause a transition to a new state.
 
-    As Taxi's initial state is a stochastic, the "p" key represents the probability of the
-    transition however this value is currently bugged being 1.0, this will be fixed soon.
-    As the steps are deterministic, "p" represents the probability of the transition which is always 1.0
+    As taxi is not stochastic, the transition probability is always 1.0. Implementing
+    a transitional probability in line with the Dietterich paper ('The fickle taxi task')
+    is a TODO.
```

Comment:
Should this TODO be here. Do we need someone to implement this?

Reply:
I've put my hand up to implement transitional probablities in taxi it once I finish the doc review. Replaces the "bugged will be fixed soon" comment :)